src/sql/parse.py

```diff
@@ -259,9 +259,12 @@ def split_args_and_sql(line):
     """
     arg_line, sql_line = line, ""
 
-    # Only separate when json arrow operators used
-    # Otherwise, behavior is unchanged
-    if not any(op in line for op in ["->", "-->"]):
+    # Filenames may include sql keywords, so we omit them
+    check = re.sub(r"('.*')", "", line)
+    check = re.sub(r'(".*")', "", check)
```

Comment:
can you explain the logic of these two lines as a comment? It's unclear what this is doing

Reply:
Please add the specific use cases also for which this logic was added. (json operators, spacing before negative sign, etc)

Reply:
[Added explanation in comments here.](https://github.com/ploomber/jupysql/pull/964/files#diff-22cfc6ec3e879941494b36e0dffaf24c100ab50173c413d821ebab94d0627ddfR262-R271)

The background behind this is in order to fix the original bug, I changed the logic for `split_args_and_sql` to split the line into an `args_line` and `sql_line` ANY time the line includes a SQL keyword. Previously it would only split the line if it included json operators `->` or `-->`, which left too many corner cases open. 

While testing this change, I noticed that queries like `%sqlplot boxplot --table 'file with spaces.csv'` were being split unintentionally, resulting in errors. This happened because 'file with spaces.csv' contains the SQL keyword `with`. Thus, I had to implement logic that would ignore any filenames when deciding if the line needed to be split, so that filenames could include words like `with`, `select`, or `begin` without being mistaken as actual SQL.